streaming/streaming.go

```diff
@@ -459,9 +459,52 @@ func LaunchDataflowJob(ctx context.Context, targetProfile profiles.TargetProfile
 			dataflowSubnetwork = fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/regions/%s/subnetworks/%s", dataflowHostProjectId, dataflowCfg.Location, dataflowCfg.Subnetwork)
 		}
 	}
-
-	launchParameters := createLaunchParameters(dataflowCfg, inputFilePattern, project, datastreamCfg, instance, dbName, streamingCfg, dataflowSubnetwork, workerIpAddressConfig)
-
+	fmt.Printf("Working with dfCfg: %v\n", dataflowCfg)
+	maxWorkers := int32(50)
+	numWorkers := int32(1)
+	if dataflowCfg.MaxWorkers != "" {
+		intVal, err := strconv.ParseInt(dataflowCfg.MaxWorkers, 10, 64)
+		if err != nil {
+			return fmt.Errorf("could not parse MaxWorkers parameter %s, please provide a positive integer as input", dataflowCfg.MaxWorkers)
+		}
+		maxWorkers = int32(intVal)
+		if maxWorkers < 1 || maxWorkers > 1000 {
+			return fmt.Errorf("maxWorkers should be a positive integer (> 0) that does not exceed 1000")
+		}
+	}
+	if dataflowCfg.NumWorkers != "" {
+		intVal, err := strconv.ParseInt(dataflowCfg.NumWorkers, 10, 64)
+		if err != nil {
+			return fmt.Errorf("could not parse NumWorkers parameter %s, please provide a positive integer as input", dataflowCfg.NumWorkers)
+		}
+		numWorkers = int32(intVal)
+		if numWorkers < 1 || numWorkers > 1000 {
+			return fmt.Errorf("numWorkers should be a positive integer (> 0) that does not exceed 1000")
+		}
+	}
+	launchParameters := &dataflowpb.LaunchFlexTemplateParameter{
+		JobName:  dataflowCfg.JobName,
+		Template: &dataflowpb.LaunchFlexTemplateParameter_ContainerSpecGcsPath{ContainerSpecGcsPath: "gs://dataflow-templates/2023-07-18-00_RC00/flex/Cloud_Datastream_to_Spanner"},
+		Parameters: map[string]string{
+			"inputFilePattern":              inputFilePattern,
+			"streamName":                    fmt.Sprintf("projects/%s/locations/%s/streams/%s", project, datastreamCfg.StreamLocation, datastreamCfg.StreamId),
+			"instanceId":                    instance,
+			"databaseId":                    dbName,
+			"sessionFilePath":               streamingCfg.TmpDir + "session.json",
+			"deadLetterQueueDirectory":      inputFilePattern + "dlq",
+			"transformationContextFilePath": streamingCfg.TmpDir + "transformationContext.json",
+		},
+		Environment: &dataflowpb.FlexTemplateRuntimeEnvironment{
+			MaxWorkers:            maxWorkers,
+			NumWorkers:            numWorkers,
+			ServiceAccountEmail:   dataflowCfg.ServiceAccountEmail,
+			AutoscalingAlgorithm:  2, // 2 corresponds to AUTOSCALING_ALGORITHM_BASIC
+			EnableStreamingEngine: true,
+			Network:               dataflowCfg.Network,
+			Subnetwork:            dataflowSubnetwork,
```

Comment:
We should add the check for validating that `subnet` region is the same as the `dataflowRegion` that we were discussing in the reverse replication PR here.

Reply:
Will take up during network config handling as a followup since this is a stopgap for zee.